libs/dh/wholesale/feature-settlement-reports-v2/src/lib/modal/dh-request-settlement-report-modal.component.ts

```diff
@@ -173,35 +186,64 @@ export class DhRequestSettlementReportModalComponent extends WattTypedModal {
 
   submitInProgress = signal(false);
 
+  // eslint-disable-next-line sonarjs/cognitive-complexity
   submit(): void {
     if (this.form.invalid) {
       return;
     }
 
-    this.requestSettlementReport()?.subscribe({
-      next: ({ loading, data }) => {
-        this.submitInProgress.set(loading);
-
-        if (loading) {
-          return;
-        }
-
-        if (this.isUpdateSuccessful(data)) {
-          this.modal()?.close(true);
-
-          this.showSuccessNotification();
-        } else {
+    this.getCalculationByGridAreas()
+      ?.pipe(takeUntilDestroyed(this.destroyRef))
+      .subscribe({
+        next: ({ settlementReportGridAreaCalculationsForPeriod }) => {
+          // If there are no calculations for all of the selected grid areas
+          if (settlementReportGridAreaCalculationsForPeriod.length === 0) {
+            return this.requestSettlementReport();
+          }
+
+          this.form.controls.calculationIdForGridAreaGroup = this.formBuilder.group({});
+
+          for (const {
+            key,
+            value: [firstElement],
+          } of settlementReportGridAreaCalculationsForPeriod) {
+            this.form.controls.calculationIdForGridAreaGroup?.addControl(
+              key,
+              new FormControl(firstElement.calculationId, { nonNullable: true })
+            );
+          }
+
+          // If there is only one calculation per selected grid area
+          const onlyOneCalculationPerSelectedGridArea =
```

Comment:
Suggestion: Would not me sense to move this up to the other guard clause that checks for length 0

Reply:
Not really because all `calculationIdForGridAreaGroup` FormControls must be created even if there's only one calculation per grid area. The values of each FormControl is then assigned to `calculationId` on line 276.